src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -257,6 +384,146 @@ WorkerFixPartitionConstraintCommandList(Oid relationId, uint64 shardId,
 }
 
 
+/*
+ * CreateFixPartitionShardIndexNamesTaskList goes over all the indexes of a distributed
+ * partitioned table, and creates the list of tasks to execute
+ * worker_fix_partition_shard_index_names UDF on worker nodes.
+ *
+ * We create parent_table_shard_count tasks,
+ * each task with parent_indexes_count x parent_partitions_count query strings.
+ */
+static List *
+CreateFixPartitionShardIndexNamesTaskList(Oid relationId)
+{
+	List *taskList = NIL;
+
+	/* enumerate the tasks when putting them to the taskList */
+	int taskId = 1;
+
+	Relation relation = RelationIdGetRelation(relationId);
+	List *indexIdList = RelationGetIndexList(relation);
+
+	/* early exit if the relation does not have any indexes */
+	if (indexIdList == NIL)
+	{
+		RelationClose(relation);
+		return NIL;
+	}
+
+	List *shardIntervalList = LoadShardIntervalList(relationId);
+
+	/* lock metadata before getting placement lists */
+	LockShardListMetadata(shardIntervalList, ShareLock);
+
+	ShardInterval *shardInterval = NULL;
+	foreach_ptr(shardInterval, shardIntervalList)
+	{
+		uint64 shardId = shardInterval->shardId;
+
+		List *queryStringList = WorkerFixPartitionShardIndexNamesCommandList(shardId,
+																			 indexIdList);
+
+		Task *task = CitusMakeNode(Task);
+		task->jobId = INVALID_JOB_ID;
+		task->taskId = taskId++;
+
+		task->taskType = DDL_TASK;
+		SetTaskQueryStringList(task, queryStringList);
+		task->dependentTaskList = NULL;
+		task->replicationModel = REPLICATION_MODEL_INVALID;
+		task->anchorShardId = shardId;
+		task->taskPlacementList = ActiveShardPlacementList(shardId);
+
+		taskList = lappend(taskList, task);
+	}
+
+	RelationClose(relation);
+
+	return taskList;
+}
+
+
+/*
+ * WorkerFixPartitionShardIndexNamesCommandList creates a list of queries that will fix
+ * all child index names of parent indexes on given shard of parent partitioned table.
+ */
+static List *
+WorkerFixPartitionShardIndexNamesCommandList(uint64 shardId, List *indexIdList)
+{
+	List *commandList = NIL;
+	Oid indexId = InvalidOid;
+	foreach_oid(indexId, indexIdList)
```

Comment:
I think I have a slight concern on the complexity of this function. 

Can we maybe move inner parts of the loops into some functions that have more explicit names? Such as

```py
def WorkerFixPartitionShardIndexNamesCommandList():
    foreach parentShardIndex:
          WorkerFixPartitionShardIndexTaskListForParentShardIndex

def WorkerFixPartitionShardIndexTaskListForParentShardIndex():
       foreach inheritedShardInex:
                 WorkerFixPartitionShardIndexTaskListForInheritedShardIndex()..
 ```              

Do you think would that help to improve the readability?

Reply:
Makes sense, that would definitely be more readable.